src/TerminalLogger.jl

```diff
@@ -104,6 +106,44 @@ function termlength(str)
     return N
 end
 
+function format_message(message, prefix, io_context)
+    formatted = sprint(show, MIME"text/plain"(), message, context=io_context)
+    msglines = split(chomp(formatted), '\n')
+    if length(msglines) > 1
+        # For multi-line messages it's possible that the message was carefully
+        # formatted with vertical alignemnt. Therefore we play it safe by
+        # prepending a blank line.
+        pushfirst!(msglines, SubString(""))
+    end
+    msglines
+end
+
+function format_message(message::AbstractString, prefix, io_context)
+    # For strings, use Markdown to do the formatting. The markdown renderer
+    # isn't very composable with other text formatting so this is quite hacky.
+    message = Markdown.parse(message)
+    prepend_prefix = !isempty(message.content) &&
+                        message.content[1] isa Markdown.Paragraph
+    if prepend_prefix
+        # Hack: We prepend the prefix here to allow the markdown renderer to be
+        # aware of the indenting which will result from prepending the prefix.
+        # Without this we will get many issues of broken vertical alignment.
+        pushfirst!(message.content[1].content, prefix)
```

Comment:
I think this will just work, but to be _super_ careful, I wonder if it makes sense to use a character from [Private Use Area](https://en.wikipedia.org/wiki/Private_Use_Areas) to create a dummy prefix:

```julia
dummy_prefix = '\uF8FF' ^ textwidth(prefix)
```

Reply:
Interesting idea, maybe a bit excessively careful. But passing the `textwidth` to `format_message` is nicer in some ways so I'll go with that.